### PR TITLE
Improve how KMT jobs are scheduled in the CI

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

* Run KMT jobs when the tasks files change
* Add `RUN_KMT_TESTS` variable to force running KMT jobs if needed.

### Motivation

Ensure that changes in those files trigger a CI run for the corresponding job.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -962,10 +967,13 @@ workflow:
   - test/new-e2e-utils/**/*
   - test/new-e2e/go.mod
   - tasks/security_agent.py
+  - tasks/kmt.py
+  - tasks/kernel_matrix_testing/*
 
 .on_security_agent_changes_or_manual:
   - <<: *if_main_branch
     allow_failure: true
+  - <<: *if_run_all_kmt_tests
```
Comment (KSerrania) [inline .gitlab-ci.yml:976]: Nit: the rules order is different in `.on_security_agent_changes_or_manual` / `on_any_kmt_test_component_or_manual` (main > all KMT tests > mergequeue) and in `.on_system_probe_or_e2e_changes_or_manual` (main > mergequeue > all KMT tests).

I think both jobs should use the main > mergequeue > all KMT tests order.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1152,6 +1163,7 @@ workflow:
 
 .on_any_kmt_test_component_or_manual:
   - <<: *if_main_branch
+  - <<: *if_run_all_kmt_tests
   - <<: *if_mergequeue
     when: never
```
Comment (KSerrania) [inline .gitlab-ci.yml:1168]:
```suggestion
  - <<: *if_mergequeue
    when: never
  - <<: *if_run_all_kmt_tests
```

**Review by KSerrania — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (gjulianm) [reply .gitlab-ci.yml]: Changed, thx!

**Pull request merged**